t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -1202,6 +1202,8 @@ test_expect_success 'sparse-index is not expanded' '
 	git -C sparse-index add README.md &&
 	ensure_not_expanded diff --staged &&
 
+	ensure_not_expanded clean -fd &&
+
```

Comment:
Could you add a test that checks that `git clean -fd` correctly removes untracked files in sparse directories?

something like:

```
git sparse-checkout set deep/deeper1 &&
mkdir deep/deeper2 folder1 &&
touch deep/deeper2/untracked &&
touch folder1/untracked &&
git clean -fd &&
test_path_is_missing deep/deeper2/untracked &&
test_path_is_missing folder1/untracked
```

I expect that the index will be expanded to do this operation, but let's make sure it happens.

Reply:
Updated the existing `git clean` tests to check those cases and explicitly verify paths are present or missing. Also, I verified that the index is expanded when untracked files are inside a sparse directory, so no tests were added to the `ensure_not_expanded` test.